histogram.go

```diff
@@ -0,0 +1,42 @@
+package imaging
+
+import (
+	"image"
+)
+
+// Histogram returns a normalized histogram of an image.
+//
+// Resulting histogram is represented as an array of 256 floats, where
+// histogram[i] is a probability of a pixel being of a particular luminance i.
+func Histogram(img image.Image) [256]float32 {
+	src := toNRGBA(img)
+	width := src.Bounds().Max.X
+	height := src.Bounds().Max.Y
+
+	var histogram [256]float32
+	var total float32
+
+	parallel(height, func(partStart, partEnd int) {
+		for y := partStart; y < partEnd; y++ {
+			for x := 0; x < width; x++ {
+				i := y*src.Stride + x*4
+
+				r := src.Pix[i+0]
+				g := src.Pix[i+1]
+				b := src.Pix[i+2]
+
+				var y float32 = 0.2126*float32(r) + 0.7152*float32(g) + 0.0722*float32(b)
```

Comment:
Why have you decided to use this formula for luminance? The image/color package from stdlib uses fractions 0.299, 0.587 and 0.114 for the gray color model. The same coefficients I used in the Grayscale function. I'm not saying one is better than the other, just curious and looking for consistency.